tiri/src/lib.rs

```diff
@@ -34,10 +46,11 @@ use yktrace::tir::{
 const DUMP_CONTEXT: usize = 4;
 
 /// Each local variable is represented by one `Value` struct.
-/// This corresponds with the Miri's `Operand` type.
 #[derive(Debug, Clone, Eq, PartialEq)]
 enum Value {
+    /// A stand-alone immediate value.
```

Comment:
Either variant of this struct are storage for local variables. Immediates are the subset not referring to a memory address.

Reply:
I must admit, that doesn't make much sense to me. Remember I'm dumb :)

Reply:
I borrowed this from Miri as it seemed a sensible way to do things. Stuff that fits in a register is an "immediate" value whereas other stuff which has to be in-memory is a different variant.

So stuff word-size integers (or smaller) are immediate values, whereas a large struct is not immediate, but "indirect".

I think that's a useful distinction to make, but I can be swayed if you disagree. This is new to me too.

Reply:
I lifted the names `Immediate` and `Indirect` from Miri, but I also know them as ( similar) addressing modes in assembler programming. I suggest we keep them.

Reply:
Here's the Intel manual's blurb on immediates:
![image](https://user-images.githubusercontent.com/604955/67866796-6516db00-fb21-11e9-90dd-ffb6b4dc73dc.png)

Their notion is little different, as they have a separate addressing mode for registers, which are a bit like local variables.

We probably don't want to obsess about CPU semantics too much though ;)

Reply:
I'm OK with the names, so long as we clearly document what they actually mean. I would never have guessed e.g. that `Immediate` means "register sized thing", so that's the key bit of information etc. etc.

Reply:
I've just realised that Miri bends the rules a little, in that immediates are typically larger than register size.

An Immeidate is either a Scalar or a ScalarPair, and each Scalar is by a u128 value with a size....

I'll find some wording that makes sense...